galsim/base.py

```diff
@@ -1462,10 +1462,13 @@ def drawKImage(self, re=None, im=None, nx=None, ny=None, bounds=None, scale=None
 
         return re,im
 
-    # Quick and dirty.  Just check reprs are equal.
-    def __eq__(self, other): return repr(self) == repr(other)
+    def __eq__(self, other):
+        if repr(self) != repr(other):
+            return False
+        return self.SBProfile == other.SBProfile
+
     def __ne__(self, other): return not self.__eq__(other)
-    def __hash__(self): return hash(repr(self))
+    def __hash__(self): return hash((repr(self), self.SBProfile.serialize()))
```

Comment:
Do you need the repr here now that you have serialize?  I would have thought making the hash using just serialize would be sufficient.